portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+
+  /*
+   * Called when a client submits a POST request to the /delete-data URL,
+   * clears database of all comments
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query =
+      Query.newEntityQueryBuilder()
+        .setKind("Comment")
+        .build();
+    QueryResults<Entity> results = datastore.run(query);
+    while(results.hasNext()) {
```

Comment:
Right now you are
- fetching all the entities in the database
- deleting them one by one

There is a better way to "delete all":
- fetch all the keys
- batch delete

This should be much more efficient since 1) you only need to read metadata (keys) instead of everything; 2) you only need to do one operation to the data store.

See https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Batch_operations for the example.

Reply:
That link was for the old API :(. For the new API, the batch delete accepts Keys as varargs. Running the query gives an iterator and it seems like the best way to convert this to varargs is by looping over it?